model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java

```diff
@@ -180,20 +184,19 @@ public boolean removeRealm(String id) {
         realm.getDefaultGroupIds().clear();
         em.flush();
 
-        int num = em.createNamedQuery("deleteGroupRoleMappingsByRealm")
-                .setParameter("realm", realm.getId()).executeUpdate();
-
         session.clients().removeClients(adapter);
 
-        num = em.createNamedQuery("deleteDefaultClientScopeRealmMappingByRealm")
+        em.createNamedQuery("deleteDefaultClientScopeRealmMappingByRealm")
                 .setParameter("realm", realm).executeUpdate();
 
         session.clientScopes().removeClientScopes(adapter);
         session.roles().removeRoles(adapter);
 
-        session.groups().getTopLevelGroupsStream(adapter).forEach(Organizations.removeGroup(session, adapter));
+        em.createNamedQuery("deleteOrganizationsByRealm")
+                .setParameter("realmId", realm.getId()).executeUpdate();
```

Comment:
I was planning to use a similar approach as other providers and call a `preRemove` method. I guess it is more consistent.

One improvement we can do in that regards is group all providers using `preRemove` to share a single hierarchy, I think. So we can call than more dynamically. But won't do that now ...

Reply:
Actually, I did this way because removing a realm should force remove organizations otherwise they will be kept in the database when the feature is disabled. We can review this once we mark the feature as supported.